documentation/development-docs/design-guidelines/module-best-practices.md

```diff
@@ -0,0 +1,193 @@
+# Module Best Practices
+
+## Module Metadata
+
+### Module Naming
+
+Unless otherwise stated, the name of the module should be similar to that of provider called in the cmdlets. For example, if the cmdlets in a module called the `Microsoft.Compute` provider, then the module should be named `Az.Compute`.
+
+### Module Manifest
+
+The module manifest file, found in the format of `Az.MyService.psd1`, contains all necessary metadata information about a module, and can be initially generated using the `New-ModuleManifest` cmdlet.
+
+Below is sample `Az.MyService.psd1` file that can be used as a template for new `.psd1` files as well. Press the "Click to expand" text below to reveal the file:
+
+<details><summary>Click to expand</summary>
+<p>
+
+```
+#
+# Module manifest for module 'Az.MyService'
+#
+# Generated by: Microsoft Corporation
+#
+# Generated on: 3/11/2019
+#
+
+@{
+
+# Script module or binary module file associated with this manifest.
+# RootModule = ''
+
+# Version number of this module.
+ModuleVersion = '0.1.0'
+
+# Supported PSEditions
+CompatiblePSEditions = 'Core', 'Desktop'
+
+# ID used to uniquely identify this module
+GUID = 'be0b7de9-01c3-4d05-adcf-848546dd70e7'
```

Comment:
Should we change this to say "get random guid using [System.Guid]::new()" instead? I don't know if it matters if multiple people have the same guid.